packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -603,12 +607,55 @@ Your $application code is in $relativeAppMain.
         ? stringArg('description')
         : 'A new Flutter plugin project.';
     templateContext['description'] = description;
+
+    final String? projectName = templateContext['projectName'] as String?;
+    final List<String> templates = <String>['plugin', 'plugin_shared'];
+    final List<String> excluded = <String>[];
+    if ((templateContext['ios'] == true || templateContext['macos'] == true) && featureFlags.isSwiftPackageManagerEnabled) {
+      templates.add('plugin_swift_package_manager');
+      templateContext['swiftLibraryName'] = projectName?.replaceAll('_', '-');
+      templateContext['swiftToolsVersion'] = minimumSwiftToolchainVersion;
+      templateContext['iosSupportedPlatform'] = SwiftPackageManager.iosSwiftPackageSupportedPlatform.format();
+      templateContext['macosSupportedPlatform'] = SwiftPackageManager.macosSwiftPackageSupportedPlatform.format();
+
+      // Exclude CocoaPod files from "plugin" template
+      final Directory pluginTemplates = templatePathProvider.directoryInPackage('plugin', globals.fs);
+      excluded.addAll(<String>[
```

Comment:
Can we rearrange the templates so these exclusions aren't necessary?  Like a `plugin_cocoapods`, `plugin_swift_package_manager`, `plugin_shared`?

Reply:
Good idea